android/src/main/java/com/bugsnag/JavaScriptException.java

```diff
@@ -30,47 +30,108 @@ public void toStream(JsonStream writer) throws IOException {
 
         writer.name("stacktrace");
         writer.beginArray();
-        for (String rawFrame : rawStacktrace.split("\\n")) {
-            writer.beginObject();
-            String[] methodComponents = rawFrame.split("@", 2);
-            String fragment = methodComponents[0];
-            if (methodComponents.length == 2) {
-                writer.name("method").value(methodComponents[0]);
-                fragment = methodComponents[1];
+
+        boolean isHermes = rawStacktrace.matches("(?s).*\\sat .* \\(.*\\d+:\\d+\\)\\s.*");
+        for (String frame : rawStacktrace.split("\\n")) {
+            if (isHermes) {
```

Comment:
Is there a scenario where a single stacktrace could contain frames from both JSCore and Hermes? If not, it'd be good to move the conditional to outside the for loop just to clarify this

Reply:
the stacktrace format is detected once for the entire thing – it's assumed that it can only contain one kind of frame. If you move the conditional out of the loop though, you have to write the loop twice?

Reply:
Completely misread this - originally thought `isHermes` was initialised inside the loop.